### ISSUE TITLE
clean up CLI code

### ISSUE THREAD

**Issue by SilentCicero** [nexusdev/dapple#268]
The CLI code is very hard to follow, even after a decent analysis. For reference, the truffle CLI code accomplished roughly the same tasks, but with a much cleaner CLI architecture. Considering the importance of the CLI, I would recommend we refactor and use cleaner CLI components. This way adding and growing dapple features and upgrades is much simpler and easier.

An example: https://github.com/ConsenSys/truffle/blob/master/cli.js -- Tim killing it here.

**Comment by mhhf** [comment]
Totally agree here. We are currently in a refactoring process to split dapple up into different modules and part of this process is to get a clean cli architecture. However, this may take some time to clean up all the cli code since we are focusing on one module at a time.
